types/block_spec.md

```diff
@@ -0,0 +1,154 @@
+# Block and Header Validity
+
+## Abstract
+
+Like all blockchains, rollups are defined as the chain of **valid** blocks from the genesis, to the head. Thus, the block and header validation rules define the chain.
+
+Verifying a block / header is done in 3 parts:
+
+1. Verify correct serialization according to the protobuf spec
+
+2. Perform basic validation of the types
+
+3. Perform verification of the new block against the previous accepted block
+
+## Basic Validation
+
+Each type contains a `.ValidateBasic()` method, which verifies that certain basic invariants hold. The `ValidateBasic()` calls are nested, starting from the `Block` struct, all the way down to each subfield.
+
+The nested basic validation is called like:
+
+```go
+Block.ValidateBasic()
+  SignedHeader.ValidateBasic()
+    Header.ValidateBasic()
+	  ProposerAddress not nil
```

Comment:
unless we allow the ABCI app to change the sequencer address, then it's possible to check the sequencer address at this step, and improve performance.

Reply:
do we definitely want to allow the ABCI app to change the sequencer address? @nashqueue